samples/csharp_dotnetcore/30.asp-mvc-bot/BotActivityControllerBase.cs

```diff
@@ -92,6 +115,59 @@ protected virtual Task OnMemberAddedAsync(ChannelAccount member, ITurnContext tu
             return Task.CompletedTask;
         }
 
+        /// <summary>
+        /// Override this method to add custom processing for each added member. For example a Welcome message.
+        /// </summary>
+        /// <param name="member">A <see cref="ChannelAccount"/> corresponding to the added member.</param>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A <see cref="Task"/> that represents the work queued to execute.</returns>
+        protected virtual Task OnMemberRemovedAsync(ChannelAccount member, ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Override this method to add custom processing for any other type of Activity not covered in this sample.
+        /// </summary>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A <see cref="Task"/> that represents the work queued to execute.</returns>
+        protected virtual Task OnEventActivityAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Override this method to add custom processing for any other type of Activity not covered in this sample.
+        /// </summary>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A <see cref="Task"/> that represents the work queued to execute.</returns>
+        protected virtual Task OnDeleteUserDataActivityAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Override this method to add custom processing for any other type of Activity not covered in this sample.
```

Comment:
```suggestion
        /// Override this method to add custom processing for a ContactRelationUpdate activity.
```
May want to add an example or explain what they might want to do in response.

Reply:
Yes I like that but I'm not sure what to say quite yet.

This is a work in progress because we want to integrate this "switch" behavior into the SDK, so I just wanted to get the low hanging fruit on the sample. At least its not a bad sample now. Thanks for opening the issue and commenting on the code review.